### ISSUE TITLE
Question regarding custom forces

### ISSUE THREAD

**Issue by tyggna** [openmm/openmm-torch#36]
Is there a way to use the OpenMM Langevin integrator, but supply my own code for the force calculations?

**Comment by peastman** [comment]
That's the whole point of this plugin.  You supply an arbitrary Pytorch model to compute the forces.  Is that not what you mean?

**Comment by tyggna** [comment]
My understanding is that this plugin takes position data, puts it into a pytorch network, and then expects an output that represents potential energy (a scalar).

I'm looking for a way to use the output of a network that also includes a Nx3 tensor output that represents force vectors on individual atoms.

**Comment by peastman** [comment]
The force is computed automatically as the gradient of the energy.  Do you want to create a non-conservative force?  What sort of functional form does it have?

**Comment by tyggna** [comment]
The network is based off an SE3 transformer, trained against ANI-2 data.  It was trained in such a way that the network itself can do the individual force calculations

**Comment by peastman** [comment]
That isn't how this plugin is designed to work, but you could modify it.  For example, here's the relevant code in the CUDA implementation.

https://github.com/openmm/openmm-torch/blob/6472507abd5127f78673cdad67edf5eca406e11d/platforms/cuda/src/CudaTorchKernels.cpp#L99-L103

It calls `forward()` and interprets the return value as energy.  Then it calls `backward()` and interprets the gradient as the force.  You could modify it to expect the force to be returned directly.

**Comment by tyggna** [comment]
Thank you, that is very helpful
